vignettes/theme.Rmd

```diff
@@ -0,0 +1,321 @@
+
+``` {.scss}
+.bs4-bright {
+  background-color: $cyan;
+  color: $orange;
+}
+```
+
+Then in your text you may use raw HTML or Pandoc syntax (as shown above):
+
+``` {.html}
+This is a [*bright*]{.bs4-bright} word.
+```
+
+``` {.html}
+::: {.bs4-bright}
+All of these words are *bright*.
+:::
+```
+
+Here is the knitted output that includes these custom rules:
+
+![](images/penguin-sass.png){.snippet}
+
+You can find a full list of variables in `vignette("bs4-variables", package = "bslib")`.
+
+When compiled, those Sass rules have access to Bootstrap Sass variables, functions, and mixins. This allows you to [write custom CSS styles in a "themeable" way](https://rstudio.github.io/bslib/articles/theming.html#a-basic-themeable-component) and/or simply leverage useful Sass functions like [`color-contrast()`](https://github.com/rstudio/bslib/blob/df10a878e0e9d1e42c83b4c2101bc8d9f00ad314/inst/sass-utils/color-contrast.scss#L9).
+
+For example, we can use the Sass [`@extend`](https://sass-lang.com/documentation/at-rules/extend) rule in our `.scss` file to bundle together multiple Bootstrap variables and tell Sass that one selector should inherit the styles of another:
+
+``` {.scss}
+.error {
+  background-color: $cyan;
+  color: $white;
+
+  &--serious {
+    @extend .error;
+    color: $warning;
+    font-size: $display4-size;
+  }
+}
+```
+
+Then you can use those rules in the body of your dashboard:
+
+``` {.html}
+::: {.error}
+Uh oh...
+:::
+
+::: {.error--serious}
+Time for a break.
+:::
+```
+
+Here is the knitted output that includes these custom rules:
+
+![](images/penguin-extend.png){.snippet}
+
+You may also reuse any [fonts](#using-fonts) specified in your YAML with SASS variables.
+
+| YAML key       | Bootstrap 4 variable     |
+|----------------|--------------------------|
+| `heading_font` | `$headings-font-family`  |
+| `base_font`    | `$font-family-base`      |
+| `code_font`    | `$font-family-monospace` |
+
+So we could set up the following rule:
+``` {.scss}
+.pkg-name {
+  background-color: $cyan;
+  color: $white;
+  font-family: $font-family-monospace;
+  padding: .5rem;
+}
+```
+
+Then use it in the body:
+
+```{.html}
+I like the [flexdashboard]{.pkg-name} package.
+```
+
+Here is the knitted output:
+
+![](images/penguin-reuse.png){.snippet}
+
+## Resources
+
+This vignette just scratches the surface of how to use Bootstrap variables and SASS together. Both are powerful tools for customizing the look and feel of your HTML documents. To learn more, see:
+
+-   Bootstrap docs: <https://getbootstrap.com/docs/4.0/>
+-   Sass docs: <https://sass-lang.com/documentation>
+-   The bslib package docs (for Bootstrap): https://rstudio.github.io/bslib/
+-   The sass package docs (for fonts): https://rstudio.github.io/sass/
+-   The thematic package docs (for theming output): https://rstudio.github.io/thematic/
```

Comment:
@apreshill Do we want to promote `https://pkgs.rstudio.com` ?

The current config already allow to access those website. Ex: https://pkgs.rstudio.com/thematic/, https://pkgs.rstudio.com/sass/